ivy_tests/test_ivy/test_functional/test_core/test_gradients.py

```diff
@@ -4,164 +4,196 @@
-        assert ivy.is_variable(var)
-    if call in [helpers.np_call, helpers.jnp_call]:
-        # numpy and jax do not support flagging variables
-        pytest.skip()
-    # value test
-    assert non_var_res is False
-    assert var_res is True
-
+@given(
+    dtype_and_x=helpers.dtype_and_values(ivy_np.valid_float_dtypes),
+    as_variable=st.booleans(),
+    native_array=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="is_variable"),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+    exclusive=st.booleans()
+)
+def test_is_variable(
+    dtype_and_x,
+    as_variable,
+    native_array,
+    num_positional_args,
+    container,
+    instance_method,
+    fw,
+    exclusive
+):
+    dtype, x = dtype_and_x
+    x = np.asarray(x,dtype=dtype)
+    helpers.test_function(
+        dtype,
+        as_variable,
+        False,
+        native_array,
+        fw,
+        num_positional_args,
+        container,
+        instance_method,
+        "is_variable",
+        x=x,
+        exclusive=exclusive
+    )
 
 # variable data
-@pytest.mark.parametrize("object_in", [[], [0.0], [1], [True], [[1.0, 2.0]]])
-@pytest.mark.parametrize("dtype", ["float16", "float32", "float64"])
-def test_variable_data(object_in, dtype, device, call):
-    if call is helpers.tf_graph_call:
-        # cannot create variables as part of compiled tf graph
-        pytest.skip()
-    if call in [helpers.mx_call] and dtype == "int16":
-        # mxnet does not support int16
-        pytest.skip()
-    if len(object_in) == 0 and call is helpers.mx_call:
-        # mxnet does not support 0-dimensional variables
-        pytest.skip()
-    # smoke test
-    var = ivy.variable(ivy.array(object_in, dtype=dtype, device=device))
-    var_data = ivy.variable_data(var)
-    # type test
-    if call is not helpers.np_call:
-        # numpy does not support variables
-        assert ivy.is_variable(var)
-        if call is not helpers.mx_call:
-            # jax variables and their data are the same instance
-            assert not ivy.is_variable(var_data, exclusive=True)
-        assert ivy.is_ivy_array(var_data)
-    # cardinality test
-    assert var_data.shape == var.shape
-    # value test
-    assert np.allclose(ivy.to_numpy(var), ivy.to_numpy(var_data))
-
+@given(
+    dtype_and_x=helpers.dtype_and_values(ivy_np.valid_float_dtypes),
+    as_variable=st.booleans(),
+    native_array=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="variable_data"),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+)
+def test_variable_data(
+    dtype_and_x,
+    as_variable,
+    native_array,
+    num_positional_args,
+    container,
+    instance_method,
+    fw
+):
+    dtype, x = dtype_and_x
+    x = np.asarray(x, dtype=dtype)
+    if fw == "torch":
+        x = torch.as_tensor(x,dtype=torch.float32)
+    if fw == "tensorflow":
+        x = tf.Variable(tf.convert_to_tensor(x,dtype=dtype))
```

Comment:
could you write a comment in this test, explaining why we need to pass these backend-specific variables as inputs to `helpers.test_function`?